cvat-core/src/server-proxy.js

```diff
@@ -619,19 +624,91 @@
-                        proxy: config.proxy,
-                    });
-                } catch (errorData) {
-                    try {
-                        await deleteTask(response.data.id);
-                    } catch (_) {
-                        // ignore
+                onUpdate('The data are being uploaded to the server 0%');
+
+                async function chunkUpload(file) {
+                    const chunkSize = 1024 * 1024 * 100; // 100 mb
+                    const totalChunks = Math.ceil(file.size / chunkSize);
+                    let currentChunkNumber = 1;
+                    async function upload() {
+                        const prevChunkNumber = currentChunkNumber - 1;
+                        const fileChunk = file.slice(prevChunkNumber * chunkSize, Math.min(prevChunkNumber * chunkSize + chunkSize, file.size));
+                        taskData.set('client_files[0]', fileChunk);
+                        taskData.set('chunk_file_name', file.name);
+                        taskData.set('end_of_upload', currentChunkNumber >= totalChunks);
+                        taskData.set('use_chunk_upload', true);
+                        try {
+                            await Axios.post(`${backendAPI}/tasks/${response.data.id}/data`, taskData, {
+                                proxy: config.proxy,
+                            });
+                            onUpdate(`The data are being uploaded to the server ${Math.round((currentChunkNumber / totalChunks) * 100)}%`);
+                            currentChunkNumber++;
+                            if (currentChunkNumber <= totalChunks) await upload();
+                        } catch (errorData) {
+                            try {
+                                await deleteTask(response.data.id);
+                            } catch (_) {
+                                // ignore
+                            }
+                            throw generateError(errorData);
+                        }
                     }
+                    await upload();
+                }
 
-                    throw generateError(errorData);
+                async function bulkUpload(files) {
+                    const requestLimit = 1024 * 1024 * 100; // 100 mb
+                    const fileGroups = [];
+                    let currentGroup = [];
+                    let currentGroupSize = 0;
+                    for (const file of files) {
+                        if (requestLimit - currentGroupSize >= file.size) {
+                            currentGroup.push(file);
+                            currentGroupSize += file.size;
+                        } else {
+                            fileGroups.push(currentGroup);
+                            currentGroupSize = file.size;
+                            currentGroup = [file];
+                        }
+                    }
+                    fileGroups.push(currentGroup);
```

Comment:
Seems like it could be rewritten with reduce in more pleasant way :)

Reply:
Done